homeassistant/components/climate/__init__.py

```diff
@@ -912,6 +912,23 @@ async def async_service_temperature_set(
     entity: ClimateEntity, service_call: ServiceCall
 ) -> None:
     """Handle set temperature service."""
+    if (
+        ATTR_TEMPERATURE in service_call.data
+        and not entity.supported_features & ClimateEntityFeature.TARGET_TEMPERATURE
+    ):
+        raise ServiceValidationError(
+            translation_domain=DOMAIN, translation_key="not_supported_temp_feature"
+        )
+    if (
+        ATTR_TARGET_TEMP_LOW in service_call.data
+        and not entity.supported_features
+        & ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+    ):
+        raise ServiceValidationError(
```

Comment:
Do we need a deprecation time for this for custom integrations?

Reply:
I'm not entirely sure as this would be quite a misuse of the entity features if someone sets it to `TARGET_TEMPERATURE` but implements handling for range.
What do you think? Go for a softer approach?

Reply:
It's just that whenever we add a validation of something like this we find that custom integrations have been doing it wrong since forever and they then break.

I'd either try try to review custom integrations in HACS and see if we find any that break by this validation or assume we need a deprecation period.

Reply:
Good point. I think it would be very tricky to try to review for this validation so I'll put in a deprecation period with logging a warning and then we change it later 👍